docs/book/beyond_basics/dependencies.md

```diff
@@ -0,0 +1,178 @@
+# Dependency Management
+
+Kubebuilder uses [dep](https://golang.github.io/dep) to manage dependencies.
+Different dependency management tasks can be done using the `dep ensure`
+command.
+
+## Adding new dependencies
+
+{% panel style="warning", title="Kubernetes Dependencies" %}
+
+Kubebuilder depends on a number of Kubernetes dependencies internally.
```

Comment:
I think this section is assuming `Kubebuilder` is referring to as `projects generated by Kubebuilder`. May be, we should make that clear.